src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -90,114 +89,148 @@ public string GetPathToResizedImage(string originalPath)
 
 			lock (this)
 			{
+				var cacheFileName = originalPath;
+
+				if (getThumbnail)
+				{
+					cacheFileName = "thumbnail_" + cacheFileName;
+				}
+
+				// if there is a cached version, return it
 				string pathToProcessedVersion;
-				if (_originalPathToProcessedVersionPath.TryGetValue(originalPath, out pathToProcessedVersion))
+				if (_originalPathToProcessedVersionPath.TryGetValue(cacheFileName, out pathToProcessedVersion))
 				{
 					if (File.Exists(pathToProcessedVersion) &&
 						new FileInfo(originalPath).LastWriteTimeUtc <= new FileInfo(pathToProcessedVersion).LastWriteTimeUtc)
 					{
 						return pathToProcessedVersion;
 					}
-					else
-					{
-						_originalPathToProcessedVersionPath.Remove(originalPath);
-					}
+
+					// the file has changed, remove from cache
+					_originalPathToProcessedVersionPath.Remove(cacheFileName);
 				}
 
-				using (var originalImage = PalasoImage.FromFile(originalPath))
+				// there is not a cached version, try to make one
+				var pathToProcessedImage = Path.Combine(_cacheFolder, Path.GetRandomFileName() + Path.GetExtension(originalPath));
+
+				if (!Directory.Exists(Path.GetDirectoryName(pathToProcessedImage)))
+					Directory.CreateDirectory(Path.GetDirectoryName(pathToProcessedImage));
+
+				// BL-1112: images not loading in page thumbnails
+				var success = false;
+				if (getThumbnail)
 				{
-					//if it's a jpeg, we don't resize, we don't mess with transparency, nothing. These things
-					//are scary in .net. Just send the original back and wash our hands of it.
-					if (ImageUtils.AppearsToBeJpeg(originalImage))
-					{
-						return originalPath;
-					}
+					// The HTML div that contains the thumbnails is 80 pixels wide, so make the thumbnails 80 pixels wide
+					success = GenerateThumbnail(originalPath, pathToProcessedImage, 80);
+				}
+				else
+				{
+					success = MakeTransparentImage(originalPath, pathToProcessedImage);
+				}
+
+				if (!success) return originalPath;
+
+				try
+				{
+					_originalPathToProcessedVersionPath.Add(cacheFileName, pathToProcessedImage); //remember it so we can reuse if they show it again, and later delete
```

Comment:
If more than one thread is accessing this dictionary, as seems likely, I'm pretty sure we should have a lock on it. I don't think dictionaries in .net are automatically thread-safe (you may want to check this), and if they're not, unpleasant things are likely to happen (occasionally and unrepeatably) if different threads try to use it at the same time.